clang/docs/checkedc/3C/README.md

```diff
@@ -48,3 +50,66 @@ that 3C will contain the core inference logic, while 5C will add
+
+We typically recommend that serious 3C users use CCI's repository to
+get 3C fixes and enhancements sooner, but in some scenarios, you may
+be better off with Microsoft's repository. Here, in detail, are the
+factors we can think of that might affect your decision (as of
+February 2021):
+
+- CCI strives to merge changes reasonably quickly from Microsoft's
+  repository to CCI's, but most 3C-specific changes are made first in
+  CCI's repository and merged to Microsoft's in batches every few
+  months. Thus, CCI's repository typically gives you a significantly
+  newer version of 3C, while Microsoft's repository typically gives
+  you a somewhat newer version of the rest of the Checked C codebase,
+  including some shared code used by 3C. The implication of that last
+  point is that a fix to the shared code in Microsoft's repository
+  would benefit Microsoft's copy of 3C immediately but would not take
+  effect on CCI's copy until the next merge from Microsoft to CCI, so
+  CCI's copy of 3C is not always newer in _all_ respects.
+
+- While the [3C regression tests](CONTRIBUTING.md#testing) are run on
+  every change to either repository, CCI's copy of 3C undergoes
+  additional testing: a set of "benchmark" tests that run nightly,
+  plus manual use by CCI engineers. If a change to CCI's repository
+  passes the regression tests but breaks the additional tests, we will
+  generally fix the problem quickly. But if the same happens in
+  Microsoft's repository, functionality of Microsoft's copy of 3C that
+  is not covered by the regression tests could remain broken for some
+  time. The problem would likely be detected the next time CCI tries
+  to merge from Microsoft, in which case we would try to get a fix
+  into Microsoft's repository reasonably quickly.
+
+- The 3C regression tests are run regularly on Windows on Microsoft's
+  repository but not on CCI's. A change that breaks the regression
+  tests on Windows generally won't be made to Microsoft's repository,
+  but such a change could be made to CCI's and the problem may not be
+  detected until the next time CCI tries to merge to Microsoft (or a
+  user reports the problem to us). So you're less likely to encounter
+  Windows-specific problems with Microsoft's repository.
+
+- On the other hand, some CCI engineers work on Mac OS X and
+  frequently run the regression tests and other manual tests on CCI's
+  copy of 3C on Mac OS X, while we are unaware of any testing of
+  Microsoft's copy of 3C on Mac OS X. So conceivably, you may be less
+  likely to encounter Mac-specific problems with CCI's repository. But
+  so far, when we've seen Mac-specific problems, we've usually gotten
+  a fix into Microsoft's repository reasonably quickly.
+
+You can potentially use some tools from one repository and some from
+the other (for example, the Checked C compiler from Microsoft's
+repository and 3C from CCI's), though neither CCI nor Microsoft tests
+this configuration.
```

Comment:
I would drop the above paragraph. It's not adding new information (this is what people would assume anyway).

Reply:
I thought you told me that someone (I forget who) was confused about this recently.  The revisions earlier in this file should help a lot, but I thought it was still best to address this point explicitly.  OK to keep the information?

Reply:
They were confused about the use of Microsoft's `checkedc` repository for the header files, but CCI's repository for `clang` and `3c`. That's a different situation than the one above, since it does not involve "tools" from two different repositories. I think all of the other changes you have made clarify what's going on sufficiently, so I'd still like to see this paragraph deleted, since I don't believe it tells the reader anything they don't know (i.e., if you use related tools from different repos, it might work, it might not ...).

Reply:
OK.  Done.  Conceivably, if a user is blocked by a bug in the Checked C compiler in our repository that has been fixed in Microsoft's repository and vice versa for 3C and doesn't feel competent to merge the fixes, it might be helpful to remind them that they can try using one tool from each repository and see if that works.  But we don't have to address this now.